engine/class_modules/sc_shaman.cpp

```diff
@@ -5303,6 +5291,13 @@ struct spiritwalkers_grace_t : public shaman_spell_t
                       options_str )
   {
     may_miss = may_crit = harmful = callbacks = false;
+    // Reduces the cooldown of Spiritwalker's Grace by ${$m1/-1000} sec
+    if ( p()->talent.graceful_spirit->ok() )
+    {
+      timespan_t reduction = timespan_t::from_seconds( p()->talent.graceful_spirit->effectN( 1 ).base_value() / -1000 );
+      cooldown->duration -= reduction;
+      p()->buff.spiritwalkers_grace->cooldown->duration -= reduction;
```

Comment:
Why does the buff have a cooldown here? Only the action should have the CD.

Reply:
i've removed the cooldown from the buff itself now which should be cleaner